racingcar-java/src/main/java/step2/StringCalculator.java

```diff
@@ -0,0 +1,55 @@
+package step2;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public class StringCalculator {
+
+    private final static String[] operators = {"+", "-", "*", "/"};
+
+    public int calculation(final String str) {
+        String[] input = isStringNullCheck(str);
+        Optional<String> optOperator = Optional.empty();
+        int result = Integer.parseInt(input[0]);
+
+        for (int i = 1; i < input.length; i++) {
+            if (isOperator(input[i])) {
+                optOperator = Optional.of(input[i]);
+            } else {
+                int y = Integer.parseInt(input[i]);
+                result = selectArithmeticOperation(optOperator.get(), result, y);
+            }
+        }
+
+        return result;
+    }
+
+    private String[] isStringNullCheck(final String str) {
+        if (str == null || str.isEmpty()) {
+            throw new IllegalArgumentException("입력 값이 비어있습니다.");
+        }
+        return str.split(" ");
+    }
+
+    private final boolean isOperator(final String str) {
+        if (Arrays.asList(operators).contains(str)) {
+            return true;
+        }
+        return false;
+    }
+
+    private int selectArithmeticOperation(final String operator, final int x, final int y) {
+        switch (operator) {
+            case "+":
+                return ArithmeticCalculator.ADD.arithmeticCalculator(x, y);
+            case "*":
+                return ArithmeticCalculator.MULTIPLE.arithmeticCalculator(x, y);
+            case "-":
+                return ArithmeticCalculator.MINUS.arithmeticCalculator(x, y);
+            case "/":
+                return ArithmeticCalculator.DIVISION.arithmeticCalculator(x, y);
+            default:
+                throw new IllegalArgumentException("연산자가 아닙니다.");
+        }
```

Comment:
switch return 문과 throw 💯 💯    
단, 객체지향적인 관점으로 말씀드리자면   
StringCalculator 가 ADD/MULTIPLE 과 같은 열거형을 알아야하는 문제점이 있겠네요!!   
인터페이스와 같은 추상화된 모듈을 사용하는 것도 실제 구현체를 모르기 위해서 사용하는 것인데  
위와 같은 실제 구현체를 알아야한다면 유지보수하기 힘들어질 수 도 있을 것 같아요   

이를 내부로 숨기는 것은 어떨까요?  

```suggestion
        ArithmeticCalculator calculator = ArithmeticCalculator.factory(operator);
        calculator.arithmeticCalculator(x, y)
```

Reply:
앗 이런 방법이 있었군요 이 부분에 대해서는 제 생각이 조금 짧았던 것 같습니다 ..! 한번 시도해보도록 하겠습니다 :) 감사합니다 !